crypto/crypto/src/ed25519.rs

```diff
@@ -76,6 +77,24 @@ impl Clone for Ed25519PrivateKey {
 #[derive(DeserializeKey, Clone, SerializeKey)]
 pub struct Ed25519PublicKey(ed25519_dalek::PublicKey);
 
+/// A private tag type that tracks if a public key is checked to protect against invalid point
```

Comment:
Would it be envisionable to put those in their own module? A [`mod`](https://doc.rust-lang.org/reference/items/modules.html) declaration would go a long way towards unifying declarations with this unified functionality.